Fix unused parameter warnings

Rombur opened the pull request.

dalg24 commented on packages/Search/test/tstLinearBVH.cpp:
```diff
@@ -515,7 +515,7 @@ std::vector<std::array<double, 3>> make_random_cloud( double Lx, double Ly,
     std::vector<std::array<double, 3>> cloud( n );
     std::default_random_engine generator;
-    std::uniform_real_distribution<double> distribution_x( 0.0, Lz );
+    std::uniform_real_distribution<double> distribution_x( 0.0, Lx );
```
I thought you got "all" of these...

Rombur replied:
me too :(

dalg24 commented on packages/Utils/test/tstMiscellaneous.cpp:
```diff
@@ -16,10 +16,10 @@
 {
     auto const dtk_version = DataTransferKit::version();
-    TEUCHOS_ASSERT( !dtk_version.empty() );
+    TEST_ASSERT( !dtk_version.empty() );
```
I had these changes sitting on some branch :)
Good catch.  Did you actually grep for `TEUCHOS_`?

Rombur replied:
No, I got a warning that `out` and `success` were not used.

dalg24 replied:
Please do so and change if there is any other instance.

Rombur replied:
I just grepped and there shouldn't be any other `TEUCHOS_ASSERT`

Rombur commented:
Don't merge this yet. I will change the `configure` script so Jenkins can catch these warnings next time

Rombur commented:
The problem is not the compiler but the flags that we are passing to gcc. These are not unused variables but unused parameters. We need `-Wextra` to get these warnings.

Rombur commented:
I have fixed more unused parameter, enable shadow warnings, and added `Wextra` but only for clang so that we are not swamped with warnings when using the docker container.

Rombur commented:
Should be good to merge. We still have about 200 warnings split equally between Kokkos, Tpetra, and Teuchos.
